### ISSUE TITLE
Field order is different in the schema

### ISSUE THREAD

**Issue by tharindu-nw** [saasquatch/json-schema-inferrer#9]
I noticed that the order of the fields in the schema is different to that of the input json object. I wanted to know whether this was intentional. If so is there a way to predict the field order in the schema? I am using this as an intermediary in one of my libraries and I am having trouble writing the tests as I cannot determine the exact output beforehand. Any help would be much appreciated.

**Comment by slisaasquatch** [comment]
The cause of this behavior is this line: https://github.com/saasquatch/json-schema-inferrer/blob/a33159525abebd33e85496eb3ca3a7a44c3f5d43/src/main/java/com/saasquatch/jsonschemainferrer/JunkDrawer.java#L165

`Collectors.toSet()` uses a `HashSet` under the hood and `HashSet` does not guarantee ordering for obvious reasons.

I guess you can say this is intentional, because 2 JSON schema objects with different field orderings should be logically identical. I also wonder why you are having trouble writing tests. I think Jackson is smart enough to know that 2 JSON objects with different field orderings are considered equal. For those reasons, I don't really feel like "fixing" the issue, even though it's a very simple fix.

```java
  public static void main(String[] args) {
    final ObjectNode o1 = JsonNodeFactory.instance.objectNode()
        .put("a", 1)
        .put("b", 2);
    final ObjectNode o2 = JsonNodeFactory.instance.objectNode()
        .put("b", 2)
        .put("a", 1);
    System.out.println(o1);
    System.out.println(o2);
    System.out.println(o1.equals(o2));
  }
```

The above main method prints

```json
{"a":1,"b":2}
{"b":2,"a":1}
true
```

**Comment by tharindu-nw** [comment]
Thank you for the quick response. I understand your reasoning and there's no need for a fix. Please label this as a question and close it.

**Comment by tcrg311** [comment]
There is a downstream effect when using this in conjunction with jsonschema2pojo, since it generates a @JsonPropertyOrder annotation using the sequence in the schema.  Using a LinkedHashSet seems like a small price to pay for the predictability.  But the tool is so immensely useful, I'm fairly sorry to even mention it.  Congratulations on some really fine work, and if the property order keeps me awake at night, I'll submit my first ever (old and wise as I am) PR.

**Comment by slisaasquatch** [comment]
Fixed in 0.1.4.

**Comment by tcrg311** [comment]
Thanks!  It must have been committed the day after I pulled.  It's a really useful utility and gives a productivity boost.  We have really huge JSON docs.
